axelor-human-resource/src/main/resources/views/LeaveLine.xml

```diff
@@ -20,8 +28,19 @@
 	    <panel-related name="leaveManagementListPanel" field="leaveManagementList" editable="true" grid-view="leave-management-grid" form-view="leave-management-form" onChange="action-leave-line-method-compute-quantity-available"/>
 	</form>
 	
+	<form name="leave-line-form-all-employee" title="Leave line" model="com.axelor.apps.hr.db.LeaveLine" width="large">
+		<panel>
+			<field name="employee" grid-view="employee-lite-grid" form-view="employee-form" />
+		</panel>
+		<panel-include view="leave-line-form"/>
+	</form>
+	
 	<action-method name="action-leave-line-method-compute-quantity-available">
 		<call class="com.axelor.apps.hr.web.leave.management.LeaveManagementController" method="computeQuantityAvailable"/>
 	</action-method>
 	
+	<action-record name="action-leave-line-record-set-name" model="com.axelor.apps.hr.db.LeaveLine">
+		<field name="name" expr="eval: leaveReason.leaveReason"/>
```

Comment:
onChange is also triggered when you empty a field, so here I get a NPE if I empty the leaveReason field.
You can simply use `?.` to empty the name when leaveReason is emptied.